libtextsecure/outgoing_message.js

```diff
@@ -293,309 +293,272 @@ OutgoingMessage.prototype = {
+              isMultiDeviceRequest = true;
+              thisDeviceMessageType = 'friend-request';
+            } else {
+              // Throttle automated friend requests
+              this.successfulNumbers.push(devicePubKey);
+              return null;
             }
           }
-        }
-
-        // Check if we need to attach the preKeys
-        let sessionCipher;
-        const isFriendRequest = thisDeviceMessageType === 'friend-request';
-        enableFallBackEncryption =
-          enableFallBackEncryption || isFriendRequest || isMultiDeviceRequest;
-        const flags = this.message.dataMessage
-          ? this.message.dataMessage.get_flags()
-          : null;
-        const isEndSession =
-          flags === textsecure.protobuf.DataMessage.Flags.END_SESSION;
-        const isSessionRequest =
-          flags === textsecure.protobuf.DataMessage.Flags.SESSION_REQUEST;
-        const signalCipher = new libsignal.SessionCipher(
-          textsecure.storage.protocol,
-          address
-        );
-        if (enableFallBackEncryption || isEndSession) {
-          // Encrypt them with the fallback
-          const pkb = await libloki.storage.getPreKeyBundleForContact(
-            devicePubKey
-          );
-          const preKeyBundleMessage = new textsecure.protobuf.PreKeyBundleMessage(
-            pkb
-          );
-          this.message.preKeyBundleMessage = preKeyBundleMessage;
-          window.log.info('attaching prekeys to outgoing message');
-        }
 
-        let messageBuffer;
-        if (isMultiDeviceRequest) {
-          const tempMessage = new textsecure.protobuf.Content();
-          const tempDataMessage = new textsecure.protobuf.DataMessage();
-          tempDataMessage.body = i18n('secondaryDeviceDefaultFR');
-          if (this.message.dataMessage && this.message.dataMessage.profile) {
-            tempDataMessage.profile = this.message.dataMessage.profile;
+          // If we're not friends with our own device then we should become friends
+          if (isOurDevice && keysFound && !isFriends) {
+            conversation.setFriendRequestStatus(
+              window.friends.friendRequestStatusEnum.friends
+            );
           }
-          tempMessage.preKeyBundleMessage = this.message.preKeyBundleMessage;
-          tempMessage.dataMessage = tempDataMessage;
-          messageBuffer = tempMessage.toArrayBuffer();
-        } else {
-          messageBuffer = this.message.toArrayBuffer();
         }
+      } catch (e) {
+        // do nothing
+      }
+    }
 
-        if (enableFallBackEncryption) {
-          sessionCipher = new libloki.crypto.FallBackSessionCipher(address);
-        } else {
-          sessionCipher = signalCipher;
-        }
-        const plaintext = this.getPlaintext(messageBuffer);
+    // Check if we need to attach the preKeys
+    const enableFallBackEncryption =
+      !keysFound || thisDeviceMessageType === 'friend-request';
```

Comment:
```suggestion
      !keysFound || thisDeviceMessageType === 'friend-request' || isMultiDeviceRequest;
```
i think you forgot this

Reply:
in fact, if you check when `isMultiDeviceRequest` is set to `true` in the old code, you will find that the `thisDeviceMessageType` is also set to `'friend-request'` at the samre time, so this is redundant

Reply:
You are correct 👍